standards/application/waku-api.md

```diff
@@ -0,0 +1,224 @@
+and for consumers to use as a single entry point to its functionalities.
+
+This API defines the RECOMMENDED interface for leveraging Waku protocols to send and receive messages. 
+Application developers SHOULD use it to access capabilities for peer discovery, message routing, and peer-to-peer reliability.
+
+## Motivation
+
+The accessibility of Waku protocols is capped by the accessibility of their implementations, and hence API.
+This RFC enables a concerted effort to draft an API that is simple and accessible, and provides an opinion on sane defaults.
+
+This effort is best done in an RFC, allowing all current implementors to review and agree on it. 
+
+The API defined in this document is an opinionated-by-purpose method to use the more agnostic [WAKU2]() protocols.
+
+## Syntax
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, 
+“RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Definitions
+
+This section defines key terms and concepts used throughout this document.
+Each term is detailed in dedicated subsections to ensure clarity and consistency in interpretation.
+
+`Multiaddr` - a self-describing format for encoding network address of a remote peer as per [libp2p addressing](https://github.com/libp2p/specs/blob/6d38f88f7b2d16b0e4489298bcd0737a6d704f7e/addressing/README.md) specification.
+
+## API design
+
+### IDL
+
+A custom Interface Definition Language (IDL) in YAML is used to define the Waku API.
+Existing IDL Such as OpenAPI, AsyncAPI or WIT do not exactly fit the requirements for this API.
+Hence, instead of having the reader learn a new IDL, we propose to use a simple IDL with self-describing syntax.
+
+An alternative would be to choose a programming language. However, such choice may express unintended opinions on the API.
+
+#### Guidelines
+
+- No `default` means that the value is mandatory
+- Primitive types are `string`, `int`, `bool`, `enum` and `uint`
+- Complex pre-defined types are:
+  - `struct`: object and other nested types
+  - `option`: a value that can be set or left null
+  - `array`: iterable object containing values of all the same type
+  - `multiaddr`: a libp2p multiaddress; may be an object or a string, most idiomatic approach depending on the language
+  - `result`: an enum type that either contain a return value (success), or an error (failure). The error is left to the implementor
+  - `error`: Left to the implementor on whether `error` types are `string` or `object` in the given language.
+- The first parameter of a function MAY be considered as the object on which a method is called. It is left to the implementor on whether an objective programming approach is idiomatic to the language.
```

Comment:
this line seems to be too specific even though it is true for `nim` and `rust`

Reply:
we can re-introduce it when we define function that needs a reference to some exiting Waku node instance.